### PR TITLE
233 refactor using react hook form on hierarchy component

### PR DESCRIPTION
**Pull request by bjornole-kodehode**
### What is missing from the codebase?

Currently, using state instead of react hook form when handling blur request

### Changes made in this pull request:

- using react hook form to handle api request 
- using a separate handler to handle snackbar api request

## Checklist

-   [x] **Linting**: Have you run the linter to ensure code consistency?
-   [x] **Code Formatting**: Is the code formatted according to project standards?
-   [x] **Review**: Have you reviewed your own changes to ensure they align with best practices?

### CODE REVIEW

**Comment by bjornole-kodehode** [comment]
- Uncertain where to put the [handleApiRequestSnackbar.ts](https://github.com/OptiCorp/inventory-app/pull/239/commits/0e582b44f9605fc32d7293eec095f0be5035447a#diff-586d09a38761925a7693594a1a0d16001faf36327aee758ca486379ff13434b2)
In a separate utility folder?

File: src/pages/partDetails/hierarchy/ChildItemSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    const { isLoading, fetchNextPage } = useGetItemsInfinite(searchTerm);
+    const [selectedChildItem, setSelectedChildItem] = useState({
+        childItemId: '',
+        childItemWpId: '',
+    });
+
+    const handleAddChildToParent = (ids: AddChildItemIds) => {
+        if (ids.childItemId === ids.itemId) {
+            setSnackbarSeverity('warning');
+            setSnackbarText('Cant add itself as child');
```
Comment (Benjamin-Lyon) [inline src/pages/partDetails/hierarchy/ChildItemSelector.tsx:62]: small fix just use an apostrophe here. "Can't" not "Cant"

**Review by Benjamin-Lyon — CHANGES_REQUESTED** [review]
Looks good 👍 Just a small spelling fix needed

Also remember that your commits should be imperative, "add this, change that, fix something" and not present tense, "adding this, changing that, fixing something".

File: src/pages/partDetails/hierarchy/ChildItemSelector.tsx (hunk 1 above)
Comment (bjornole-kodehode) [reply src/pages/partDetails/hierarchy/ChildItemSelector.tsx]: [99bdf6d](https://github.com/OptiCorp/inventory-app/pull/239/commits/99bdf6d56e585c17801ac2dd1dae1d5c3a1a2382)

**Review by Benjamin-Lyon — APPROVED** [review]

**Pull request merged**